rotkehlchen/chain/ethereum/modules/juicebox/decoder.py

```diff
@@ -0,0 +1,127 @@
+            )
+        except RemoteError:
+            log.error(f'Failed to query ipfs hash for juicebox project {project_id}')
+            return None
+
+        metadata_url = JUICEBOX_IPFS_GATEWAY + ipfs_hash
+        try:
+            log.debug(f'Querying juicebox metadata at {metadata_url}')
+            metadata: dict[str, Any] = requests.get(
+                url=metadata_url,
+                timeout=CachedSettings().get_timeout_tuple(),
+            ).json()
+        except requests.RequestException as e:
+            log.error(f'Failed to query juicebox project {project_id} metadata due to {e}')
+            return None
+
+        return metadata.get('name')
+
+    def _decode_pay(self, context: DecoderContext) -> DecodingOutput:
+        """Decode pay with rewards in juicebox"""
+        if context.tx_log.topics[0] != PAY_SIGNATURE:
+            return DEFAULT_DECODING_OUTPUT
+
+        try:
+            topic_data, decoded_data = decode_event_data_abi_str(context.tx_log, PAY_ABI)
+        except DeserializationError as e:
+            log.error(
+                f'Failed to deserialize Juicebox event at '
+                f'{context.transaction.tx_hash.hex()} due to {e}',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = token_normalized_value_decimals(
+            token_amount=decoded_data[2],
+            token_decimals=18,  # it is always ETH
+        )
+        project_id = topic_data[2]
+        project_name = self._query_project_name(project_id=project_id)
+        is_donation = project_id == LEGAL_DEFENSE_TORNADO_PROJECT_ID
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_ETH and
+                event.balance.amount == amount and
+                event.location_label == decoded_data[0]
+            ):
+                event.counterparty = CPT_JUICEBOX
+                if is_donation:
+                    action_verb = 'Donated'
+                    event.event_subtype = HistoryEventSubType.DONATE
+                else:
+                    action_verb = 'Paid'
+
+                event.notes = f'{action_verb} {amount} ETH at Juicebox'
+                if project_name is not None:
+                    event.notes += f' to project {project_name}'
+                if len(decoded_data[4]) != 0:
+                    event.notes += f' with memo: "{decoded_data[4]}"'
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                event.counterparty = CPT_JUICEBOX
+                action_verb = 'donating' if is_donation else 'contributing'
+                event.notes = f'Received a commemorative NFT for {action_verb} via Juicebox'
```

Comment:
Since it's not just donations we can't assume NFTs are just commemorative. Some may have utility after all.

```suggestion
                event.notes = f'Receive an NFT for {action_verb} via Juicebox'
```